collect_app/src/test/java/org/odk/collect/android/preferences/stores/AdminPreferencesDataStoreTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.odk.collect.android.preferences.stores
+
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.odk.collect.android.preferences.PreferencesDataSource
+import org.odk.collect.android.preferences.PreferencesDataSourceProvider
+
+class AdminPreferencesDataStoreTest {
+    private lateinit var adminPreferencesDataStore: AdminPreferencesDataStore
+    private lateinit var preferencesDataSourceProvider: PreferencesDataSourceProvider
+    private lateinit var adminPreferences: PreferencesDataSource
+
+    @Before
+    fun setup() {
+        preferencesDataSourceProvider = mock(PreferencesDataSourceProvider::class.java)
+        adminPreferences = mock(PreferencesDataSource::class.java)
+        `when`(preferencesDataSourceProvider.getAdminPreferences()).thenReturn(adminPreferences)
+        adminPreferencesDataStore = AdminPreferencesDataStore(preferencesDataSourceProvider)
+    }
+
+    @Test
+    fun `When putString() called, should save() be called on adminPreferences`() {
```

Comment:
I think for test names like this you can omit "when". I'd usually try and save it (or "given") for if I'm describing a condition. So for instance I'd name this test `putString saves string in adminPreferences`.  If the method exploded when you passed it a `null` key I'd write a test like `putString, when key is null, throws exception`. I'm also not including "should" here as it's not really needed. 

This is massively subjective so interested in your thoughts here! I think we have a nice opportunity to discuss different styles and then add some thoughts to our `CODE-GUIDELINES.md`.

Reply:
I agree, Fixed.